massa-pos-exports/src/pos_final_state.rs

```diff
@@ -706,23 +706,17 @@ impl PoSFinalState {
 
         match range.start_bound() {
             Included(slot) => {
-                self.deferred_credits_serializer
-                    .slot_ser
-                    .serialize(slot, &mut start_key_buffer)
-                    .expect(DEFERRED_CREDITS_SER_ERROR);
+                start_key_buffer.extend_from_slice(&slot.to_bytes_key());
             }
             Excluded(slot) => {
-                self.deferred_credits_serializer
-                    .slot_ser
-                    .serialize(
-                        &slot
-                            .get_next_slot(self.config.thread_count)
-                            .expect(DEFERRED_CREDITS_SER_ERROR),
-                        &mut start_key_buffer,
-                    )
-                    .expect(DEFERRED_CREDITS_SER_ERROR);
+                start_key_buffer.extend_from_slice(
+                    &slot
+                        .get_next_slot(self.config.thread_count)
+                        .expect(DEFERRED_CREDITS_SER_ERROR)
+                        .to_bytes_key(),
+                );
             }
-            _ => {}
+            Unbounded => {}
         };
 
         for (serialized_key, serialized_value) in db.iterator_cf(
```

Comment:
what about
```rust
        let iterator_cf = db.iterator_cf(
            STATE_CF,
            MassaIteratorMode::From(&start_key_buffer, MassaDirection::Forward),
        ).filter(|(k, _)| k.starts_with(DEFERRED_CREDITS_PREFIX.as_bytes()));

        for (serialized_key, serialized_value) in
        iterator_cf
        {
            // if !serialized_key.starts_with(DEFERRED_CREDITS_PREFIX.as_bytes()) {
            //     // println!("break !start_with");
            //     break;
            // }
```
if I'm correct, it's equivalent and based on few measures the loop is 10% faster

Reply:
isn't there a risk that it iterates starting from `start_key_buffer` and until **the end of the whole database** ? I'm afraid `filter()` is not short-circuiting, contrary to `https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.take_while`

Reply:
Yes, `take_while` is the correct version of what I wanted to do.
In my test, I counted the number of entries at the end and they were the same. So I was wrong, but the conditions were favorable so the timing looks correct.
I tried again with `take_while` and it seems to be a bit faster (probably because it short-circuits)

Reply:
In another PR, we may want to go over similar behaviour in the code and see if it's worth it to change. (/!\ some of the DB iteration have weird logic, e.g. the iter over async_messages in a comment above, we have to make sure that everything is the same before changing it).